src/test/java/api/loans/RenewalAPITests.java

```diff
@@ -1180,6 +1181,206 @@ public void testRespectSelectedTimezoneForDueDateCalculations() throws Exception
+      .getId();
+
+    checkRenewalAttempt(expectedDueDate, dueDateLimitedPolicyId);
+  }
+
+  @Test
+  public void canRenewalForSystemDateWhenSystemDateFallWithinRangeDueDateLimit() throws
+    InterruptedException,
+    ExecutionException,
+    TimeoutException,
+    MalformedURLException {
+
+    FixedDueDateSchedulesBuilder fixedDueDateSchedules = new FixedDueDateSchedulesBuilder()
+      .withName("Fixed Due Date Schedule")
+      .addSchedule(wholeMonth(2019, DateTimeConstants.MARCH))
+      .addSchedule(todayOnly());
+
+    DateTime expectedDueDate = DateTime.now(DateTimeZone.UTC)
+      .withTimeAtStartOfDay()
+      .withHourOfDay(23)
+      .withMinuteOfHour(59)
+      .withSecondOfMinute(59);
+
+    final UUID fixedDueDateSchedulesId = loanPoliciesFixture.createSchedule(
+      fixedDueDateSchedules).getId();
+
+    LoanPolicyBuilder currentDueDateRollingPolicy = new LoanPolicyBuilder()
+      .withName("System Date Rolling Policy")
+      .rolling(Period.months(1))
+      .limitedBySchedule(fixedDueDateSchedulesId)
+      .renewFromSystemDate();
+
+    UUID dueDateLimitedPolicyId = loanPoliciesFixture.create(currentDueDateRollingPolicy)
+      .getId();
+
+    checkRenewalAttempt(expectedDueDate, dueDateLimitedPolicyId);
+  }
+
+  @Test
+  public void cannotRenewalForRollingLoanWhenDueDatNotFallWithinRangeDueDateLimit() throws
+    InterruptedException,
+    ExecutionException,
+    TimeoutException,
+    MalformedURLException {
+
+    DateTime futureDateTime = DateTime.now(DateTimeZone.UTC).plusMonths(1);
+
+    FixedDueDateSchedulesBuilder fixedDueDateSchedules = new FixedDueDateSchedulesBuilder()
+      .withName("Fixed Due Date Schedule in the Future")
+      .addSchedule(wholeMonth(futureDateTime.getYear(), futureDateTime.getMonthOfYear()));
+
+    final UUID fixedDueDateSchedulesId = loanPoliciesFixture.createSchedule(
+      fixedDueDateSchedules).getId();
+
+    LoanPolicyBuilder currentDueDateRollingPolicy = new LoanPolicyBuilder()
+      .withName("System Date Rolling Policy")
+      .rolling(Period.months(1))
+      .limitedBySchedule(fixedDueDateSchedulesId)
+      .renewFromSystemDate();
+
+    UUID dueDateLimitedPolicyId = loanPoliciesFixture.create(currentDueDateRollingPolicy)
+      .getId();
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    DateTime loanDueDate =
+      new DateTime(2019, DateTimeConstants.APRIL, 21, 11, 21, 43);
+
+    loansFixture.checkOutByBarcode(smallAngryPlanet, jessica, loanDueDate);
+
+    useLoanPolicyAsFallback(
+      dueDateLimitedPolicyId,
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.activeNotice().getId()
+    );
+
+    loansFixture.attemptRenewal(422, smallAngryPlanet, jessica);
+  }
+
+  @Test
+  public void  canRenewalForCurrentDueDateWhenDueDatNotFallWithinRangeAlternateDueDateLimited()
```

Comment:
My understanding is that renewal should fail when the current due date does not fall within a schedule?

What am I missing about this scenario?

Reply:
As per test this is positive case, changed test method name